system/core/events.php

```diff
@@ -256,55 +256,38 @@ function createGotoPageAndExecuteCommandButton($label, $langPageView, $command,
 		return '<input type="button" class="button" value="'.$label.'" onclick="hypha(\''.$langPageView.'\', \''.$command.'\', \''.$argument.'\');" />';
 	}
 
-	function getCsrfToken() {
-		if (!isset($_SESSION['hyphaCsrfToken'])) {
-			session_start();
-			regenerateCsrfToken();
-			session_write_close();
-		}
-
-		return $_SESSION['hyphaCsrfToken'];
-	}
-
-	/*
-		Function: regenerateCsrfToken
-
-		Regenerate the CSRF token. Should be called when a new
-		session starts, such as during login. Should be called
-		while the session is already open (e.g. between
-		session_start() and session_write_close()).
-	*/
-	function regenerateCsrfToken() {
-		$_SESSION['hyphaCsrfToken'] = bin2hex(openssl_random_pseudo_bytes(8));
-	}
-
 	// Automatically insert the CSRF token into all forms in the
 	// generated document
 	registerPostProcessingFunction('injectCsrf');
 	function injectCsrf(HTMLDocument $html) {
+		global $O_O;
 		$forms = $html->find('form');
 		foreach ($forms as $form) {
 			// if form does not have a csrf field, inject csrf field.
 			if ($form->find('input[name=csrfToken]')->count() === 0) {
 				$form->append($input = $html->create('<input name="csrfToken" type="hidden"'));
-				$input->setAttr('value', getCsrfToken());
+				$input->setAttr('value', $O_O->getCsrfToken());
 			}
 		}
 	}
 
 	// execute posted commands
-	function executePostedCommand() {
-		if(isset($_POST['command'])) {
-			if (!isset($_POST['csrfToken']) || $_POST['csrfToken'] != getCsrfToken()) {
+	function executePostedCommand($O_O) {
+		static $processed = false;
```

Comment:
Would it make sense to add this global/static var to `$O_O`?
Feels weird to have such a state variable lying around :-)

Reply:
Hm, I'm not so sure. This sneakily hidden state is not ideal, exposing this state globally through `$O_O` does not seem correct either: This is really an implementation detail of `executePostedCommand`. I think it would be good to just keep it like this now, and fix this in a future refactor (the way post commands are handled is expected to change anyway, once all datatypes are converted to the "new style").

Reply:
Cool, just wanted to double check :-)